orchagent/nexthopkey.h

```diff
@@ -98,6 +117,73 @@ struct NextHopKey
     {
         return (ip_address.getV4Addr() == 0);
     }
+
+    bool isMplsNextHop() const
+    {
+        return (label_stack.getSize() > 0);
+    }
+
+    std::string parseMplsNextHop(const std::string& str)
+    {
+        std::string ip_str;
+        auto keys = tokenize(str, LABELSTACK_DELIMITER);
+        if (keys.size() == 1)
+        {
+            // No MPLS info to parse
+            ip_str = str;
+        }
+        else if (keys.size() == 3)
+        {
+            // Expected MPLS format = "<outsegtype>+<labelstack>+<non-mpls-str>"
+            // key[0] = <outsegtype> = "swap" | "pop"
```

Comment:
How do we specify number of pop labels in this format?

Reply:
There is a new key 'mpls_pop' in the DB where num labels are specified. For pop operation, should we specify both 'outsegtype' and 'mpls_pop' in the DB?

Reply:
Number of labels to POP is property of INSEG_ENTRY. It's property of label route object

Reply:
"pop" is a typo... should be "push".  typo is fixed.  NextHopKey only describes MPLS outseg.  Pop info is part of MPLS inseg (see saimpls.h) and would not be included in nexthopkey.